src/navigation/partial_gate.py

```diff
@@ -0,0 +1,102 @@
+from dataclasses import dataclass
+import numpy as np
+
+from util import np_utils
+from typing import Optional
+from state import BaseState
+from trajectory import Trajectory
+from drive import get_drive_command
+from aenum import Enum, NoAlias
+from context import Context
+
+STOP_THRESH = 0.2
+DRIVE_FWD_THRESH = 0.95
+POST_SEPARATION = 2  # meters
+
+
+@dataclass
+class PartialGateTrajectory(Trajectory):
+    def partial_gate_traj(post_pos, rover_pos):
+        """
+        Generates a trajectory to find second of two posts
+        :param post_pos:    position of the post (np.ndarray)
+        :param rover_pos:   position of the rover (np.ndarray). Assumes that the rover is facing the post
+        :return:            list of positions for the rover to traverse List(np.ndarray)
+        """
+        # If the post is at (a, b) and rover is at (a + x, b + y),
+        # we'd calculate rover-to-post as c * (-x, -y).Then,
```

Comment:
might wanna add in this comment that `c` is the post separation, otherwise its kinda confusing